napalm_junos/junos.py

```diff
@@ -399,6 +399,21 @@ def get_environment(self):
                     elif structured_temperature_data['yellow-alarm'] <= temp:
                         environment_data['temperature'][sensor_object]['is_alert'] = True
 
+        # Try to correct Power Supply information
+        pem_table = dict()
+        try:
+            power_supplies.get()
+        except RpcError:
+            # Not all platforms have support for this
```

Comment:
What platforms have support for this?

Reply:
I don't know... I only know it works for a QFX10k but it does not work in the qfx5110-48s-4c. I do not have access to other platforms to test that...
```
user@juniper> show chassis power 
error: command is not valid on the qfx5110-48s-4c
```
According to https://www.juniper.net/documentation/en_US/junos/topics/reference/command-summary/show-chassis-power.html this command is supported in the MX (same output), also
supported in the PTX (but the output is different) and EX9200

Reply:
Alright, that tends to be very obscure... I check on MX and it is fine.